src/InsertionsClient/Api/Props/Models/PropsFile.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Net.Insertions.Models;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace Microsoft.Net.Insertions.Props.Models
+{
+    /// <summary>
+    /// This type represents a props file that contains the variables
+    /// to be used during VS build process.
+    /// </summary>
+    public class PropsFile
+    {
+        /// <summary>
+        /// Path to this file on disk.
+        /// </summary>
+        public string Path { get; private set; }
+
+        /// <summary>
+        /// Has content of this file changes since we loaded it?
+        /// </summary>
+        public bool isDirty { get; private set; }
+
+        private XDocument? _xDocument { get; set; }
+
+        private bool _isLoaded => _xDocument != null;
+
+        /// <summary>
+        /// Matches a variable assignement line.
+        /// First matched group is the variable name, second one is the value.
+        /// </summary>
+        /// <example>NetCoreSharedHostVersion=3.1.2;</example>
+        private static Regex _variableAssignmentRegex = new Regex(@"\b(\w*)=([\w|.|-]*);");
+
+        /// <summary>
+        /// Creates an instance of PropsFile.
+        /// </summary>
+        /// <param name="path">Path to the file on disk.</param>
+        public PropsFile(string path)
+        {
+            Path = path;
```

Comment:
There are a couple of reasons why I didn't test the path:
- Most importantly: the path doesn't come from user. We get it from `System.IO.Directory.GetFiles` method. Unless the file was deleted after we acquired it and before we load it, this shouldn't be a concern.
- About checking for existance: We will potentially have many props file instances, but very few will be loaded depending on which packages were updated. Checking for existence of each file will result in an unnecessary system call for most of the files. Instead, we load the file when necessary, which already performs this check
- Value shouldn't be null since we have nullable references enabled.